tsl/src/bgw_policy/job.c

```diff
@@ -89,14 +89,65 @@ get_chunk_id_to_reorder(int32 job_id, Hypertable *ht)
 															 -1);
 }
 
+static int64
+get_compression_window_end_value(Dimension *dim, Jsonb *config)
+{
+	Oid partitioning_type = ts_dimension_get_partition_type(dim);
+
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		int64 lag = policy_compression_get_older_than_int(config);
+		Oid now_func = ts_get_integer_now_func(dim);
+		ts_interval_now_func_validate(now_func, partitioning_type);
+
+		if (InvalidOid == now_func)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("integer_now function must be set")));
+
+		return ts_time_value_to_internal(ts_interval_from_now_func_get_datum(lag,
+																			 partitioning_type,
+																			 now_func),
+										 partitioning_type);
+	}
+	else
+	{
+		Datum res = TimestampTzGetDatum(GetCurrentTimestamp());
+		Interval *lag = policy_compression_get_older_than_interval(config);
+
+		switch (partitioning_type)
+		{
+			case TIMESTAMPOID:
+				res = DirectFunctionCall1(timestamptz_timestamp, res);
+				res = DirectFunctionCall2(timestamp_mi_interval, res, IntervalPGetDatum(lag));
+
+				return ts_time_value_to_internal(res, partitioning_type);
+			case TIMESTAMPTZOID:
+				res = DirectFunctionCall2(timestamptz_mi_interval, res, IntervalPGetDatum(lag));
+
+				return ts_time_value_to_internal(res, partitioning_type);
+			case DATEOID:
+				res = DirectFunctionCall1(timestamptz_timestamp, res);
+				res = DirectFunctionCall2(timestamp_mi_interval, res, IntervalPGetDatum(lag));
+				res = DirectFunctionCall1(timestamp_date, res);
+
+				return ts_time_value_to_internal(res, partitioning_type);
+			default:
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("unknown time type OID %d", partitioning_type)));
+				pg_unreachable();
+		}
+	}
+}
+
 static int32
-get_chunk_to_compress(Hypertable *ht, FormData_ts_interval *older_than)
+get_chunk_to_compress(Hypertable *ht, Jsonb *config)
 {
 	Dimension *open_dim = hyperspace_get_open_dimension(ht->space, 0);
 	StrategyNumber end_strategy = BTLessStrategyNumber;
-	Oid partitioning_type = ts_dimension_get_partition_type(open_dim);
-	int64 end_value = ts_time_value_to_internal(ts_interval_subtract_from_now(older_than, open_dim),
-												partitioning_type);
+
+	int64 end_value = get_compression_window_end_value(open_dim, config);
```

Comment:
I think new line should distinguish declaration block from return call, not between declarations (to improve readability):
```suggestion
	int64 end_value = get_compression_window_end_value(open_dim, config);
	
```